File: pronunciapa_client/tool/verify_ipa_payload.dart
import 'dart:io';
import '../lib/domain/entities/ipa_cli.dart';

void main(List<String> args) {
  final path = args.isNotEmpty ? args.first : '../outputs/ipa_practice_es_s.json';
  final file = File(path);
  if (!file.existsSync()) {
    stderr.writeln('File not found: $path');
    exit(1);
  }
  final raw = file.readAsStringSync();
  final payload = parseIpaCliPayload(raw);
  if (payload == null) {
    stderr.writeln('Invalid IPA payload');
    exit(1);
  }
  if (payload is IpaPracticeSetPayload) {
    stdout.writeln('kind=${payload.kind} items=${payload.items.length}');
    if (payload.items.isNotEmpty) {
      stdout.writeln('first=${payload.items.first.text}');
    }
  } else if (payload is IpaExplorePayload) {
    stdout.writeln('kind=${payload.kind} examples=${payload.examples.length}');
    if (payload.examples.isNotEmpty) {
      stdout.writeln('first=${payload.examples.first.text}');
    }
  } else {
    stdout.writeln('kind=${payload.kind}');
  }
}

File: scripts/verify_ipa_payload.js
const fs = require('node:fs');

const path = process.argv[2] || 'outputs/ipa_practice_es_s.json';
const raw = fs.readFileSync(path, 'utf8');
const payload = JSON.parse(raw);

const kind = payload?.kind;
if (!kind) {
  throw new Error('IPA payload missing kind');
}

const items = Array.isArray(payload.items) ? payload.items : [];
const examples = Array.isArray(payload.examples) ? payload.examples : [];
const total = items.length || examples.length;

console.log(`kind=${kind} total=${total}`);
if (payload.sound?.ipa) {
  console.log(`sound=${payload.sound.ipa}`);
}
if (items[0]?.text) {
  console.log(`first=${items[0].text}`);
} else if (examples[0]?.text) {
  console.log(`first=${examples[0].text}`);
}
